backend/protos/xyz/block/ftl/v1/language/language.proto

```diff
@@ -15,11 +136,138 @@ message Error {
   }
 
   string msg = 1;
-  schema.Position pos = 2;
-  int64 endColumn = 3;
   ErrorLevel level = 4;
+  Position pos = 5;
+}
+
+message Position {
+  string filename = 1;
+  int64 line = 2;
+  int64 startColumn = 3;
+  int64 endColumn = 4;
 }
 
 message ErrorList {
   repeated Error errors = 1;
 }
+
+// Request to build a module.
+message BuildRequest {
+  // The root path for the FTL project
+  string project_path = 1;
+  // Indicates whether to watch for file changes and automatically rebuild
+  bool rebuild_automatically = 2;
+
+  BuildContext build_context = 3;
+}
+
+// AutoRebuildStarted should be sent when the plugin decides to start rebuilding automatically.
+//
+// It is not required to send this event, though it helps inform the user that their changes are not yet built.
+// FTL may ignore this event if it does not match FTL's current build context and state.
+// If the plugin decides to cancel the build because another build started, no failure or cancellation event needs
+// to be sent.
+message AutoRebuildStarted {
```

Comment:
I was thinking that it might be good to have a sequence diagram in the design doc for these flows, WDYT?